bob-backend/src/pearl/group.rs

```diff
@@ -15,6 +15,7 @@ pub type HoldersContainer =
 #[derive(Clone, Debug)]
 pub struct Group {
     holders: Arc<RwLock<HoldersContainer>>,
+    reinit_lock: Arc<RwLock<()>>,
```

Comment:
I think we can add this lock directly to the `holder`. But to do this, you need to make sure that the functions called inside do not try to re-acquire this lock